2-Draft-Accepted/RFCxxxx-PowerShellGet-3.0.md

```diff
@@ -0,0 +1,394 @@
+    "AzureAutomation"
+  ]
+}
+```
+
+An example cache with 5000 resources (approximately the number of unique packages
+currently published to PowerShellGallery.com) is ~700KB in compressed json form.
+
+The cache would have both latest stable and latest prerelease versions of resources.
+
+>[!NOTE]
+>Need to experiment if it makes sense to have a single cache file for all repositories
+>or a different cache per repository for size and perf reasons.
+>Perf tests will determine if the cache needs to be in binary form.
+
+### Automatic updating of the cache
+
+On any operation that reaches out to a repository, a REST API will be called to
+see if the hash of the cache matches the current cache and if not, a new one
+is downloaded.
+
+>[!NOTE]
+>In the implementation we'll need to handle the case where the module attempts
+>to update the cache while it is being read.
+
+If the repository doesn't support this new API, it falls back to current behavior
+in PSGet v2.
+This means that there is no local cache of that repository and operations will
+always connect to that repository.
+
+### Repository management
+
+`Register-PSResourceRepository` will allow for registering additional repositories.
+A `-PSGallery` switch enables registering PSGallery should it be accidentally removed.
+The `-URL` will accept the HTTP address without the need to specify `/api/v3` as
+that will be assumed and discovered at runtime (trying v3 first, then falling
+back to v2, then the literal URL).
+Support for local filesystem repositories must be maintained.
+A `-Trusted` switch indicates whether to prompt the user when installing resources
+from that repository.
+By default, if this switch is not specified, the repository is untrusted.
+A `-Repositories` parameter will accept an array of hashtables equivalent to
+the parameter names (like splatting).
+
+```powershell
+Register-PSResourceRepository -Repositories @(
+  @{ URL = "https://one.com"; Name = "One"; Trusted = $true; Credential }
+  @{ URL = "https://powershellgallery.com"; Name = "PSGallery"; Trusted = $true; Default = $true }
+)
+```
+
+A `-Name` parameter allows for setting a friendly name.
+
+A `-Priority` parameter allows setting the search order of repositories.
+A lower value has higher priority.
+If not specified, the default value is 50.
+PSGallery will have a default value of 25.
+
+`Get-PSResourceRepository` will list out the registered repositories.
+
+`Set-PSResourceRepository` can be used to update a repository URL, trust level,
+or priority.
+
+`Unregister-PSResourceRepository` can be used to un-register a repository.
+
+### Finding resources
+
+Current design of PowerShellGet is to have different cmdlets for each of the
+different resources types that it supports:
+
+`Find-Command`
+`Find-DscResource`
+`Find-Module`
+`Find-RoleCapability`
+`Find-Script`
+
+Instead, these will be abstracted by a `Find-PSResource` cmdlet.
+A `-Type` parameter will accept an array of types to allow filtering.
```

Comment:
I generally prefer the different commands because their calls are pre-optimised for returning results, and i feel that the -Type is adding additional complexity into a command that could already end up being complex anyway

Reply:
I'm optimizing for users who don't care about the type of resource, they just want to find something related to `AD` (for example).

Reply:
Shorten it to just `Find-Resource`.  People can call it via long-form `PSGet/Find-Resource` if they want to be super clear.

`Find-Resource` is not currently a taken command name.

Reply:
`Resource` as a noun seems a bit too generic, this is specifically a resource for PowerShell.  I'll add it to the `Considerations` section and will let Committee debate it.